### ISSUE TITLE
Is it possible to use a pipe instead of a ".pov" file ? 

### ISSUE THREAD

**Issue by Zulko** [POV-Ray/povray#35]
Hi there ! I have seen a doc on the internet where it is said that if you provide option "+I-" on the command line you can pipe the pov code to povray without needing to write it to a file. However it doesn't seem to work: I get this error whatever I try:

```
Possible Parse Error: Cannot find file '-', even after trying to append file
 type extension.
Parse Error: Cannot open input file.
Fatal error in parser: Cannot parse input.
```

I am using POV-Ray 3.7 which I compiled from this repo a few days ago.

Is this option deprecated  ? Thanks in advance !

**Comment by byoungb** [comment]
Yeah what is up with this... found a few posts that made it seem like this worked. And I just wasted way too long trying to get it to work before finding this. Very disappointed.

Persistence of Vision(tm) Ray Tracer Version 3.7.0.unofficial (g++ 4.9.2 @x86_64-redhat-linux-gnu)

**Comment by ccann** [comment]
Any movement on this?

**Comment by wfpokorny** [comment]
I'm not aware of any work as yet to support this anew. My recollection is it was a dos only feature originally. 

If you are in an environment where you have the bash shell available, you might be able to hack to something useful with a command along the lines of: 

bash -c 'PID=$$; cat aoi.pov>this${PID}.pov; povray +Ithis${PID}.pov -D -P +A0.1 +wt1; rm this${PID}.pov;'
  
where the 'cat aoi.pov' part would be replaced with whatever is creating your *.pov input. Multiples of the above command can run at one time each getting a reasonably unique process ID for each.  If many renderings going at once,  controlling the thread/cpu-core use with +wt<n> might be a good idea as normally POV-Ray looks to use all the threads available.

**Comment by c-lipka** [comment]
> Any movement on this?

Until quite recently, implementing the requested feature would have posed a fundamental problem, since POV-Ray scenes aren't compiled nor pre-parsed in any way, and any macro invocation or loop in the scene would have required to rewind to an earlier position in the scene file, which is impossible with input from a pipe.

However, the feature is within reasonable reach now, thanks to a mechanism implemented a few months ago (for a different feature) to parse POV-Ray scene code from memory. By having POV-Ray simply load the entire (main) scene file into memory first, and only parse it once it's fully loaded, we could indeed accept arbitrary scene files from standard input.

In other words, it is now solely a matter of priorities and manpower.
